python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -20,179 +20,142 @@
 
 import lsst.afw.image as afwImage
 from lsst.afw.geom import SpanSet
 from lsst.geom import Point2I
+import lsst.log
 import lsst.afw.detection as afwDet
 
-__all__ = ["LsstSource", "LsstHistory"]
-
-
-class BlankSource(Component):
-    """Placeholder for a source that cannot be initialized
-
-    In DM-19790 this class will be removed when skipped sources are
-    not deblended at all.
+__all__ = ["init_source", "morphToHeavy", "modelToHeavy"]
+
+logger = lsst.log.Log.getLogger("meas.deblender.deblend")
+
+
+def init_source(frame, peak, observation, bbox,
+                symmetric=False, monotonic=True,
+                thresh=5, components=1):
+    """Initialize a Source
+
+    The user can specify the number of desired components
+    for the modeled source. If scarlet cannot initialize a
+    model with the desired number of components it continues
+    to attempt initialization of one fewer component until
+    it finds a model that can be initialized. If all of the
+    models fail, including a `PointSource` model, then skip
+    the source.
+
+    Parameters
+    ----------
+    frame: `LsstFrame`
+        The model frame for the scene
+    peak: `PeakRecord`
+        Record for a peak in the parent `PeakCatalog`
+    observation: `LsstObservation`
+        The images, psfs, etc, of the observed data.
+    bbox: `Rect`
+        The bounding box of the parent footprint.
+    symmetric: `bool`
+        Whether or not the object is symmetric
+    monotonic: `bool`
+        Whether or not the object has flux monotonically
+        decreasing from its center
+    thresh: `float`
+        Fraction of the background to use as a threshold for
+        each pixel in the initialization
+    components: int
+        The number of components for the source.
+        If `components=0` then a `PointSource` model is used.
     """
-    def __init__(self, frame, center):
-        self.pixel_center = center
-        sed = np.zeros((frame.C,))
-        morph = np.zeros((frame.Ny, frame.Nx))
-        super().__init__(frame, sed=sed, morph=morph, fix_sed=True, fix_morph=True)
-
-    def update(self):
-        return self
+    assert components <= 2
+    xmin = bbox.getMinX()
+    ymin = bbox.getMinY()
+    center = np.array([peak.getIy()-ymin, peak.getIx()-xmin], dtype=int)
+
+    while components > 1:
+        try:
+            source = MultiComponentSource(frame, center, observation, symmetric=symmetric,
+                                          monotonic=monotonic, thresh=thresh)
+            if (np.any([np.isnan(c.sed) for c in components]) or
+                    np.all([c.sed <= 0 for c in source.components])):
+                logger.warning("Could not initialize")
+                raise ValueError("Could not initialize source")
+        except Exception:
+            # If the MultiComponentSource failed to initialize
+            # try an ExtendedSource
+            components -= 1
```

Comment:
Not really. See previous comment, with the addendum that this happens quite often for faint sources and there is no need to overload the logger with useless warnings, since 90% of the time the warning would be "this source has a SNR too low to initialize as an `ExtendedSource`.